File: app/src/main/java/africa/of/designers/churchill/Adapter/LessonAdapter.java
package africa.of.designers.churchill.Adapter;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;


import java.util.List;

import africa.of.designers.churchill.LessonDetailActivity;
import africa.of.designers.churchill.R;
import africa.of.designers.churchill.model.Lessons;

public class LessonAdapter extends RecyclerView.Adapter<LessonAdapter.MyViewHolder> {

    Context context ;
    List<Lessons> mData;


    public LessonAdapter(Context context, List<Lessons> mData) {
        this.context = context;
        this.mData = mData;
    }


    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {


        View view = LayoutInflater.from(context).inflate(R.layout.item_movie,viewGroup,false);
        return new MyViewHolder(view);


        }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, int i) {

        myViewHolder.TvTitle.setText(mData.get(i).getTitle());
        myViewHolder.ImgCourse.setImageResource(mData.get(i).getThumbnail());

    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {


        private TextView TvTitle;
        private ImageView ImgCourse, LockedImg;


        public MyViewHolder(@NonNull final View itemView) {

            super(itemView);
            TvTitle = itemView.findViewById(R.id.item_movie_title);
            ImgCourse = itemView.findViewById(R.id.item_movie_img);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {

                    Lessons mov = mData.get(getAdapterPosition());

                    context.startActivity(new Intent(context, LessonDetailActivity.class)
                            .putExtra("title", mov.getTitle())
                            .putExtra("imgURL", mov.getThumbnail())
                            .putExtra("imgCover", mov.getCoverPhoto())
                            .putExtra("imgDesc", mov.getDescription()));

                }
            });

        }
    }
}

File: app/src/main/java/africa/of/designers/churchill/ui/adapter/LessonAdapter.kt
package africa.of.designers.churchill.ui.adapter

import africa.of.designers.churchill.R
import africa.of.designers.churchill.data.model.Lesson
import africa.of.designers.churchill.ui.MainListener
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_lesson.view.*
import kotlinx.android.synthetic.main.slide_item.view.*

class LessonAdapter: RecyclerView.Adapter<LessonAdapter.MyHolder>() {
    private var items = ArrayList<Lesson>()
    private lateinit var mOnClickListener: MainListener

    fun setListData(context: Context, data: ArrayList<Lesson>) {
        this.items = data
        mOnClickListener = context as MainListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        val inflater = LayoutInflater.from(parent.context).inflate(R.layout.item_lesson, parent, false)

        return MyHolder(inflater, items, mOnClickListener)
    }

    override fun onBindViewHolder(holder: MyHolder, position: Int) {

        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class MyHolder(view: View, items: ArrayList<Lesson>, mOnClickListener: MainListener): RecyclerView.ViewHolder(view) {
        private val lessonImage: ImageView = view.item_lesson_img
        private val lessonTitle: TextView = view.item_lesson_title

        init {
            view.setOnClickListener {
                mOnClickListener.onGoToLessonDetail(items[adapterPosition])
            }
        }

        fun bind(data: Lesson) {
            lessonTitle.text = data.title
            lessonImage.setImageResource(data.thumbnail)
        }
    }

}
